build/README.md

```diff
@@ -646,7 +650,7 @@ Cleans up your custom cluster by reseting Helm.
 
 ## Dependencies
 
-This project uses the [dep](https://github.com/golang/dep) as a dependency manager. You can see the list of dependencies [here](https://github.com/GoogleCloudPlatform/agones/blob/master/Gopkg.toml).
+This project uses the [go modules](https://github.com/golang/go/wiki/Modules) as its manager. You can see the list of dependencies [here](https://github.com/GoogleCloudPlatform/agones/blob/master/go.mod).
```

Comment:
(Assuming I understand this correctly) - do we want to put a section in here explaining that we are using vendoring, and that requires the use of `go mod vendor` to save dependency changes  (https://tip.golang.org/cmd/go/#hdr-Make_vendored_copy_of_dependencies)

..and maybe we should add something about how we have some vendored patches for apimachinery? :man_shrugging:  WDYT?

Reply:
It wouldn't hurt and would be more clear for people reading through so I am all for it.

Reply:
Fair enough. I'll write up a small ticket for it - was just thinking out loud. 

Otherwise the rest of this LGTM!

Reply:
Done.